kinder/pkg/cri/host/archive.go

```diff
@@ -185,6 +199,44 @@ func editManifestRepositories(raw []byte, editRepositories func(string) string)
 	return json.Marshal(entries)
 }
 
+// https://github.com/opencontainers/image-spec/blob/main/image-index.md
+type index struct {
+	SchemaVersion int `json:"schemaVersion"`
+	Manifests     []struct {
+		MediaType   string            `json:"mediaType"`
+		Digest      string            `json:"digest"`
+		Size        int               `json:"size"`
+		Annotations map[string]string `json:"annotations"`
+		Platform    struct {
+			Architecture string `json:"architecture"`
+			OS           string `json:"os"`
+		} `json:"platform"`
+	} `json:"manifests"`
+}
+
+func editIndexRepositories(raw []byte, editRepositories func(string) string) ([]byte, error) {
+	var idx index
+	if err := json.Unmarshal(raw, &idx); err != nil {
+		return nil, err
+	}
+
+	for _, m := range idx.Manifests {
+		const containerdImage = "io.containerd.image.name"
```

Comment:
The annotation `io.containerd.image.name` is not a part of the standard oci image spec. All pre-Defined Annotation Keys can be found [here](https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys). There is no `io.containerd.image.name` in the list. but oci image spec allows for arbitrary annotations. so the annotation is added by containerd (may be buildkit).

`org.opencontainers.image.ref.name` is the pre-defined annotation for name of the reference for a target. 

In containerd, `io.containerd.image.name` stores the fully qualified reference name of a image. it includes the registry, repository, tag. but `org.opencontainers.image.ref.name` stores only the tag of the image.

```
{
  "annotations": {
    "io.containerd.image.name": "docker.io/library/alpine:latest",
    "org.opencontainers.image.ref.name": "latest"
  }
}
```

I'm not sure whether the `org.opencontainers.image.ref.name` can be used to store a fully qualified reference name of a image. The image spec does not say anything about it. 

```
{
  "annotations": {
    "org.opencontainers.image.ref.name": "docker.io/library/alpine:latest"
  }
}
```

But it is possible to store the fully qualified reference name of a image because the validation of the annotation allows this. The rules for the annotation value are:

```
ref       ::= component ("/" component)*
component ::= alphanum (separator alphanum)*
alphanum  ::= [A-Za-z0-9]+
separator ::= [-._:@+] | "--"
```

As the OCI spec was based on the Docker image format, the containerd may more closely follow the oci image spec. However, in my mind, I'm not sure whether there is a certain reason for containerd to use `io.containerd.image.name` instead of `org.opencontainers.image.ref.name`, maybe it is just a historical reason or a design decision.

In kinder, only getting the image name from the annotation `io.containerd.image.name` makes sense.

Just for the record, the containerd gets the image name from annotaions in the following order:

```
1. io.containerd.image.name if present
2. org.opencontainers.image.ref.name

code: https://github.com/containerd/containerd/blob/main/client/import.go#L257
```

cc @neolit123

Reply:
this is what i tested and it worked. the annotation was in the yaml and if it was patched the image was imported with a different name with ctr, which is what we want.

the bigger question is why it happened. i got some indication on slack it happened because the docker version used to produce the tars was updated in k/release. but then it was reverted. the people who did all that did not reply.

this pr is currently not needed and can be closed. hopefully sig release will write an email about such spec changing in the future on the k-dev mailing list.